qiskit_experiments/base_experiment.py

```diff
@@ -113,7 +112,7 @@ def run(self, backend, experiment_data=None, **kwargs):
 
         # Create new experiment data
         if experiment_data is None:
-            experiment_data = self.__experiment_data__(self)
+            experiment_data = self.__experiment_data__(backend, self._type)
```

Comment:
Need to pass options (e.g. `share_level`) to the experiment data constructor, coming from the `run` method kwargs.